src/utils/ipv6.ts

```diff
@@ -0,0 +1,67 @@
+function decompress(ipv6: string) {
+
+    ipv6 = ipv6.replace('[', '').replace(']','');
+
+    if (!validate(ipv6)) {
+       throw new Error('Invalid address: ' + ipv6);
+    }
+    ipv6 = ipv6.toLowerCase()
+ 
+    const nh = ipv6.split(/\:\:/g);
+    if (nh.length > 2) {
+       throw new Error('Invalid address: ' + ipv6);
+    }
+ 
+    let sections = [];
+    if (nh.length == 1) {
+       sections = ipv6.split(/\:/g);
+       if (sections.length !== 8) {
+          throw new Error('Invalid address: ' + ipv6);
+       }
+    } else if (nh.length == 2) {
+       const n = nh[0];
+       const h = nh[1];
+       const ns = n.split(/\:/g);
+       const hs = h.split(/\:/g);
+       for (let i in ns) {
+          sections[i] = ns[i];
+       }
+       for (let i = hs.length; i > 0; --i) {
+          sections[7 - (hs.length - i)] = hs[i - 1];
+       }
+    }
+    for (let i = 0; i < 8; ++i) {
+       if (sections[i] === undefined) {
+          sections[i] = '0000';
+       }
+       sections[i] = leftPad(sections[i], '0', 4);
+    }
+    return sections.join(':');
+ };
+
+function validate(ip: string) {
+    return /^[a-f0-9\\:]+$/ig.test(ip);
+ };
+ 
+ function leftPad(section: string, padd: string, times: any) {
+    const padding = padd.repeat(times);
+    if (section.length < padding.length) {
+        section = padding.substring(0, padding.length - section.length) + section;
+    }
+    return section;
+ };
+
+ export function matchIPV6(orignalIp: string, ipv6: string) {
```

Comment:
Let's compare IPv6 addresses in this function only. Rename arguments accordingly.